tobago-example/tobago-example-demo/src/main/ts/demo-sidebar.ts

```diff
@@ -0,0 +1,484 @@
+  }
+
+  connectedCallback(): void {
+    // Cache all section elements first to avoid repeated DOM queries
+    this.sectionElements = Array.from(document.querySelectorAll<HTMLElement>("tobago-section[id^='page:mainForm:']"));
+
+    // Build the section tree hierarchy once
+    this.buildSectionTree();
+
+    // Render the initial tree
+    this.renderContentTree();
+
+    // Add event listeners for page changes
+    window.addEventListener("hashchange", this.hashChangeHandler);
+    window.addEventListener("resize", this.resizeHandler);
+
+    // Use ResizeObserver instead of window resize for better performance
+    this.setupResizeObserver();
+
+    // Initial position adjustment
+    this.adjustFixedPosition();
+
+    // Initial scroll to hash if present with proper timing
+    if (window.location.hash) {
+      requestAnimationFrame(() => {
+        this.handleHashChange();
+        this.updateActiveSection();
+      });
+    } else {
+      this.updateActiveSection();
+    }
+  }
+
+  disconnectedCallback(): void {
+    // Clean up event listeners
+    window.removeEventListener("hashchange", this.hashChangeHandler);
+    window.removeEventListener("resize", this.resizeHandler);
+
+    // Clean up resize observer
+    if (this.resizeObserver) {
+      this.resizeObserver.disconnect();
+      this.resizeObserver = null;
+    }
+
+    // Clear any pending timeouts
+    if (this.scrollThrottleTimeout !== null) {
+      window.clearTimeout(this.scrollThrottleTimeout);
+      this.scrollThrottleTimeout = null;
+    }
+  }
+
+  /**
+   * Throttle function to limit execution frequency
+   */
+  private throttle(func: Function, delay: number): () => void {
+    let lastCall = 0;
+    return function(): void {
+      const now = new Date().getTime();
+      if (now - lastCall >= delay) {
+        lastCall = now;
+        func();
+      }
+    };
+  }
+
+  /**
+   * Set up resize observer for more efficient layout adjustments
+   */
+  private setupResizeObserver(): void {
+    if ("ResizeObserver" in window) {
+      this.resizeObserver = new ResizeObserver(this.throttle(() => {
+        this.adjustFixedPosition();
+      }, 100));
+
+      // Observe header elements that might affect positioning
+      const header = document.querySelector("header");
+      const navbar = document.querySelector(".navbar");
+
+      if (header) this.resizeObserver.observe(header);
+      if (navbar) this.resizeObserver.observe(navbar);
+    }
+  }
+
+  /**
+   * Adjusts the fixed position based on current header height
+   * Using requestAnimationFrame to batch layout reads/writes
+   */
+  private adjustFixedPosition(): void {
+    requestAnimationFrame(() => {
+      try {
+        // Batch all DOM reads first
+        const headerHeight = document.querySelector("header")?.clientHeight || 0;
+        const navbarHeight = document.querySelector(".navbar")?.clientHeight || 0;
+        const fixedTopHeight = document.querySelector(".fixed-top")?.clientHeight || 0;
```

Comment:
Warum so viele Variablen für den Header? Der gesuchte Header is "tobago-header.sticky-top".
Für Header/Footer kann man gut einen Getter anlegen.

Außerdem: Ich finde offsetHeight besser als clientHeight, weil offsetHeight auch die Border berücksichtigt.